src/ui_fsmenu/addons.cc

```diff
@@ -1487,33 +1538,158 @@ struct RemoteInteractionWindow : public UI::Window {
+			bar->set_total(most_votes);
+			bar->set_state(nr_votes);
+			bar->set_show_percent(false);
+
+			UI::Textarea* label =
+			   new UI::Textarea(box, UI::PanelStyle::kFsMenu, UI::FontStyle::kFsMenuLabel,
+			                    std::to_string(vote), UI::Align::kCenter);
+
+			box->add(bar, UI::Box::Resizing::kExpandBoth);
+			box->add_space(kRowButtonSpacing);
+			box->add(label, UI::Box::Resizing::kFullSize);
+			voting_stats_.add(box, UI::Box::Resizing::kExpandBoth);
+			voting_stats_.add_inf_space();
+		}
+
+		box_votes_.add(&voting_stats_summary_, UI::Box::Resizing::kFullSize);
+		box_votes_.add_space(kRowButtonSpacing);
+		box_votes_.add(&voting_stats_, UI::Box::Resizing::kExpandBoth);
+		box_votes_.add_space(kRowButtonSpacing);
+		box_votes_.add(&own_voting_, UI::Box::Resizing::kFullSize);
+		box_votes_.add_space(kRowButtonSpacing);
+
+		tabs_.add("comments", (boost::format(_("Comments (%u)")) % info_.user_comments.size()).str(),
+		          &box_comments_);
+
+		if (nr_screenshots_) {
+			tabs_.add("screenshots",
+			          (boost::format(_("Screenshots (%u)")) % info_.screenshots.size()).str(),
+			          &box_screenies_);
+			tabs_.sigclicked.connect([this]() {
+				if (tabs_.active() == 1) {
+					next_screenshot(0);
+				}
+			});
+		} else {
+			box_screenies_.set_visible(false);
+		}
+
+		tabs_.add(
+		   "votes", (boost::format(_("Votes (%u)")) % info_.number_of_votes()).str(), &box_votes_);
+
+		main_box_.add(&tabs_, UI::Box::Resizing::kExpandBoth);
+		main_box_.add_space(kRowButtonSpacing);
+		main_box_.add(&ok_, UI::Box::Resizing::kFullSize);
+
+		screenshot_next_.set_enabled(nr_screenshots_ > 1);
+		screenshot_prev_.set_enabled(nr_screenshots_ > 1);
+		screenshot_cache_.resize(nr_screenshots_, nullptr);
+		screenshot_next_.sigclicked.connect([this]() { next_screenshot(1); });
+		screenshot_prev_.sigclicked.connect([this]() { next_screenshot(-1); });
+
+		main_box_.set_size(get_inner_w(), get_inner_h());
 	}
 
 private:
-	UI::Box box_;
+	void next_screenshot(int8_t delta) {
+		assert(nr_screenshots_ > 0);
+		while (delta < 0) {
+			delta += nr_screenshots_;
+		}
+		current_screenshot_ += delta;
+		current_screenshot_ %= nr_screenshots_;
+		assert(current_screenshot_ < static_cast<int32_t>(screenshot_cache_.size()));
+
+		auto it = info_.screenshots.begin();
+		std::advance(it, current_screenshot_);
+
+		screenshot_stats_.set_text(
+		   (boost::format(_("%1$u / %2$u")) % (current_screenshot_ + 1) % nr_screenshots_).str());
+		screenshot_descr_.set_text(it->second);
+		screenshot_.set_tooltip("");
+
+		if (screenshot_cache_[current_screenshot_]) {
+			screenshot_.set_icon(screenshot_cache_[current_screenshot_]);
+			return;
+		}
+
+		const Image* image = nullptr;
+		const std::string screenie =
+		   parent_.net().download_screenshot(info_.internal_name, it->first);
+		if (!screenie.empty()) {
+			image = g_image_cache->get(screenie);
+			g_fs->fs_unlink(screenie);
```

Comment:
Better leave the image in the temp directory and try it before downloading. Then the cache also works after closing the screenshot window. The temp directory could be cleaned up on exiting the game.